tests/test_poly.py

```diff
@@ -0,0 +1,287 @@
+    assert (unpacked_test_2 == unpacked_test).all()
+
+def test_poly_mean():
+    #Zero'th order polynomial fitting should be pretty trivial, just
+    #the same as the mean fitting. So much of this code is just taken from
+    #the mean testing in test_simple
+    N = 0
+    nparam = 5
+    orders = [N for i in xrange(nparam)]
+    interp = piff.Polynomial(orders)
+    nstars = 100
+
+    #Choose some random values of star parameters
+    vectors = [ numpy.random.random(size=nparam) for i in range(nstars) ]
+
+    #take the mean of them. Our curve fit should be able to reproduce this.
+    mean = numpy.mean(vectors, axis=0)
+
+    #Choose some random positions in the field.
+    #NB think more about bounds and 
+    pos = [ (numpy.random.random()*10, numpy.random.random()*10)
+            for i in range(nstars) ]
+
+    # Run our solver.
+    interp.solve(pos, vectors)
+
+    #we expect one set of coefficients per object
+    assert len(interp.coeffs)==5
+
+    #We should have very close values (not necessarily identical) since
+    #we calculate these in numerically different ways.
+    for mu, val in zip(mean, interp.coeffs):
+        assert numpy.isclose(mu, val[0,0])
+
+    #We also expect that if we interpolate to any point we just
+    #get the mean as well
+    for i in xrange(30):
+        p=(numpy.random.random()*10, numpy.random.random()*10)
+        v = interp.interpolate(p)
+        assert numpy.allclose(v, mean)
+
+
+def sub_poly_linear(type1):
+    #Now lets do something more interesting - test a linear model.
+    #with no noise this should fit really well, though again not
+    #numerically perfectly.
+    numpy.random.seed(12834)
+    nparam = 3
+    N = 1
+    nstars=50   
+    orders = [N for i in xrange(nparam)]
+    interp = piff.Polynomial(orders, poly_type=type1)
+    X = 10.0 #size of the field
+    Y = 10.0
+
+    pos = [ (numpy.random.random()*X, numpy.random.random()*Y)
+            for i in range(nstars) ]
+
+
+    #Let's make a function that is linear just as a function of one parameter
+    #These are the linear fit parameters for each parameter in turn
+    m1 = numpy.random.uniform(size=nparam)
+    m2 = numpy.random.uniform(size=nparam)
+    c = numpy.random.uniform(size=nparam)
+    def f(pos):
+        u = pos[0]
+        v = pos[1]
+        r = m1*u+m2*v+c
+        return r
+
+    #Simulate the vectors under this model
+    vectors = [f(p) for p in pos]
+
+    #Fit them. Linear fitting is quite easy so this should 
+    #be okay
+    interp.solve(pos, vectors)
+
+    #Check that the interpolation recovers the desired function
+    for i in xrange(30):
+        p=(numpy.random.random()*X, numpy.random.random()*Y)
+        assert numpy.allclose(f(p), interp.interpolate(p))
+
+def test_poly_linear():
+    for poly_type in PolynomialsTypes:
+        sub_poly_linear(poly_type)
+
+def test_poly_quadratic():
+    #This is basically the same as linear but with
+    #quadratic variation
```

Comment:
Any reason not to do this the same way as linear, with all the types in `PolynomialsTypes`?